### PR TITLE
Implement running site-specific tests

### PR DESCRIPTION
**Pull request by arjan**
Implements #1331; also added some docs on how to run the tests and what tests look like.

(based off of `0.x` branch but I'll cherrypick to master once merged)

### CODE REVIEW

File: src/filewatcher/z_filewatcher_inotify.erl [hunk 1]
```diff
@@ -117,7 +117,7 @@ handle_info(timeout, State) ->
     {noreply, State};
 
 handle_info(_Info, State) ->
-    ?DEBUG(_Info),
+    %%?DEBUG(_Info),
```
Comment (ddeboer) [inline src/filewatcher/z_filewatcher_inotify.erl]: Just remove this? :wink:

**Comment by ddeboer** [comment]
This is awesome! 🌴 ☀️

File: doc/developer-guide/testing.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+It is possible to create end-to-end integration tests for Zotonic
+websites.  Tests like these are called *sitetests*. They run within
+the Zotonic shell, starting the website under test using a special
+database schema so that the main database is not affected by the
+tests.
+
+To run the sitetests for a site called ``example``, run the following command::
+
+  z_sitetest:run(example).
```
Comment (ddeboer) [inline doc/developer-guide/testing.rst]: Nitpicking: please indent code samples 4 spaces.

File: doc/developer-guide/testing.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+This will stop the ``example`` site, create a new database schema
+(called ``z_sitetest``), start the site (which installs all site data
+into the new schema), and then scans all compiled Erlang modules for
```
Comment (ddeboer) [inline doc/developer-guide/testing.rst]: Perhaps link to [module versioning](http://zotonic.com/docs/latest/developer-guide/modules.html#module-versioning) here.

File: doc/developer-guide/testing.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+modules named ``example_*_sitetest.erl``. These found test modules are
+then run using the standard EUnit test framework.
+
+
+Example `sitetest` module
```
Comment (ddeboer) [inline doc/developer-guide/testing.rst]: reST needs double backticks.

File: doc/developer-guide/testing.rst [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+   administrator_name_test() ->
+     ?assertEqual(<<"Site Administrator">>, m_rsc:p(1, title, sudo_context())),
+     ok.
+
+
+Development
+-----------
+
+Running the sitetest tests are integrated into the filewatcher
```
Comment (ddeboer) [inline doc/developer-guide/testing.rst]: s/are/is

File: doc/developer-guide/testing.rst [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+hooks. As soon as you edit a ``*sitetest.erl`` file, all tests in that
+specific sitetest file will be executed.
```
Comment (ddeboer) [inline doc/developer-guide/testing.rst]: Wow, cool feature!

File: src/support/z_sitetest.erl [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+-author("Arjan Scherpenisse <arjan@miraclethings.nl>").
+
+-export([run/1, run/2, find_sitetest_modules/1]).
+
+-include_lib("zotonic/include/zotonic.hrl").
+-include_lib("epgsql/include/pgsql.hrl").
+
+%% @doc Run all _sitetest tests for the given host
+run(Host) when is_atom(Host) ->
+    run(Host, find_sitetest_modules(Host)).
```
Comment (ddeboer) [inline src/support/z_sitetest.erl]: Why do you name this `Host` instead of `Site`? It seems the latter is more usual in Zotonic?

Comment (arjan) [inline src/support/z_sitetest.erl]: Agreed. However in the `#context{}` record it's called `host`. I was confused myself about this, actually. I'll rename this. We should also consider renaming `host` to `site` in the `#context{}`, imho.
cc @mworrell

**Comment by ddeboer** [comment]
Submitted a PR against this one to fix the db connection params: https://github.com/arjan/zotonic/pull/1.

File: src/support/z_sitetest.erl (hunk 7 above)
Comment (mworrell) [inline src/support/z_sitetest.erl]: Yes, let's change this in the master. Consistent naming is important.

So for now on we will be talking about a `site` running on Zotonic. Which is much more consistent with the `sites` directory anyway. I will make a change request for this.

Comment (arjan) [inline src/support/z_sitetest.erl]: See #1333

**Comment by ddeboer** [comment]
Tested this with arjan#1 merged in, and then it works fine. @arjan Please merge that PR, so we can merge this one. :wink:

**Comment by ddeboer** [comment]
Merged. @arjan Can you cherry-pick this to master?

**Pull request merged**